proxy/ProxySession.cc

```diff
@@ -313,3 +315,21 @@ ProxySession::support_sni() const
 {
   return _vc ? _vc->support_sni() : false;
 }
+
+PoolableSession *
+ProxySession::protocol_creation(NetVConnection *netvc)
+{
+  // Figure out what protocol was negotiated
+  int proto_index             = SessionProtocolNameRegistry::INVALID;
+  SSLNetVConnection *sslnetvc = dynamic_cast<SSLNetVConnection *>(netvc);
```

Comment:
Let's not depend on SSLNetVC. `ALPNSupport` should be enough here.

Reply:
Actually, I don't see any reason to receive a NetVC here. Just a protocol id is enough. Let's deal with NetVC on HttpSM side.

How about this? I want function names to have verbs.
```
PoolableSession *
ProxySession::create_outbound_sesion(int protocol_id) {
  auto iter = ProtocolSessionCreateMap.find(protocol_id);
   ink_release_assert(iter != ProtocolSessionCreateMap.end());
   return iter->second();
}
```